shopfloor_reception/services/reception.py

```diff
@@ -983,6 +1020,35 @@ def scan_line(self, picking_id, barcode):
             return handler(picking, search_result.record)
         return self._scan_line__fallback(picking, barcode)
 
+    def _find_related_move_in_picking(self, picking, line):
+        move = line.move_id
+        if move.picking_id != picking:
+            move = fields.first(
+                picking.move_ids.filtered(lambda m: m.product_id == line.product_id)
+            )
+        return move
+
+    def scan_line_repeat(self, picking_id, last_processed_line_id):
+        """Repeat the previous line operation on the same move."""
```

Comment:
an extended docstring explaining what is the expected result would be nice.

Reply:
Added, will squash soon